src/CameraControls.ts

```diff
@@ -723,6 +723,8 @@ export class CameraControls extends EventDispatcher {
 			if ( ! this._domElement ) return;
 			if ( ! this._enabled || this.mouseButtons.wheel === ACTION.NONE ) return;
 
+			this._resetFocalOffsets();
```

Comment:
What about moving this line to the beginning of `_zoomInternal()` and `_dollyInternal()` with `if ( this.dollyToCursor )`?
Then wheel zoom/dolly will reset focalOffsets.

eg
```typescript
protected _dollyInternal = ( delta: number, x: number, y : number ): void => {

  if ( this.dollyToCursor ) this._resetFocalOffsets();
```